File: src/main/java/view25d/prims/UpdateOneTurtleView.java
package view25d.prims;

import org.nlogo.api.AgentSet;
import org.nlogo.api.Argument;
import org.nlogo.api.Context;
import org.nlogo.api.DefaultCommand;
import org.nlogo.api.ExtensionException;
import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;

import view25d.View25DExtension;
import view25d.view.TurtleView;

public class UpdateOneTurtleView extends DefaultCommand {

	@Override
	public Syntax getSyntax() {
		int[] argType = {Syntax.StringType(), Syntax.AgentsetType()};
		return  Syntax.commandSyntax( argType );
	}
	
	@Override
	public String getAgentClassString() {
		return "O";
	}

	@Override
	public void perform(Argument[] args, Context context) throws ExtensionException, LogoException {
		String title = args[0].getString();
		AgentSet as = args[1].getAgentSet();
		if ( View25DExtension.turtleWindowMap.containsKey( title ) ) {
			TurtleView win = View25DExtension.turtleWindowMap.get( title );
			win.setAgentSet( as );
			win.manuallyRefreshReporterView(context);
		} else {
			throw new ExtensionException("No Variable Visualization window with title "+title);
		}
	}

}

File: src/main/scala/prims/UpdateOneTurtleView.scala
package view25d.prims

import org.nlogo.api.{ Argument, Command, Context, ExtensionException }
import org.nlogo.core.Syntax

import view25d.View25DExtension

class UpdateOneTurtleView extends Command {
  override def getSyntax: Syntax =
    Syntax.commandSyntax(right = List(Syntax.StringType, Syntax.AgentsetType), agentClassString = "O---")

  override def perform(args: Array[Argument], context: Context): Unit = {
    val title = args(0).getString

    View25DExtension.turtleWindowMap.get(title) match {
      case Some(window) =>
	  	window.setAgentSet(args(1).getAgentSet)
	  	window.manuallyRefreshReporterView(context)
      case None => throw new ExtensionException(s"No Variable Visualization window with title $title");
    }
  }
}

